cloud-gcp/src/main/java/com/sequenceiq/cloudbreak/cloud/gcp/compute/GcpInstanceResourceBuilder.java

```diff
@@ -447,17 +447,11 @@ private CloudVmInstanceStatus stopStart(GcpContext context, AuthenticatedContext
         String instanceId = instance.getInstanceId();
         try {
             Get get = compute.instances().get(projectId, availabilityZone, instanceId);
-            String state = stopRequest ? "RUNNING" : "TERMINATED";
             Instance instanceResponse = get.execute();
-            if (state.equals(instanceResponse.getStatus())) {
-                Operation operation = stopRequest ? compute.instances().stop(projectId, availabilityZone, instanceId).setPrettyPrint(true).execute()
-                        : executeStartOperation(projectId, availabilityZone, compute, instanceId, instance.getTemplate(), instanceResponse.getDisks());
-                CloudInstance operationAwareCloudInstance = createOperationAwareCloudInstance(instance, operation);
-                return new CloudVmInstanceStatus(operationAwareCloudInstance, InstanceStatus.IN_PROGRESS);
-            } else {
-                LOGGER.debug("Instance {} is not in {} state - won't start it.", state, instanceId);
-                return null;
-            }
+            Operation operation = stopRequest ? compute.instances().stop(projectId, availabilityZone, instanceId).setPrettyPrint(true).execute()
```

Comment:
Can we use If condition here instead of Elvis operator and please add some additional log that identifies which operation should be initiated?

Reply:
updated